app/src/main/java/org/piwigo/ui/main/AlbumsFragment.java

```diff
@@ -106,7 +107,7 @@ public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
         binding.albumRecycler.setHasFixedSize(true);
         binding.albumRecycler.setLayoutManager(new GridLayoutManager(getContext(), calculateColumnCount()));
         binding.photoRecycler.setHasFixedSize(true);
-        binding.photoRecycler.setLayoutManager(new GridLayoutManager(getContext(), calculateColumnCount() * 3));
+        binding.photoRecycler.setLayoutManager(new GridLayoutManager(getContext(), calculateColumnCount() * Integer.parseInt(SettingsPreferences.getSettingPreference(SettingsPreferences.KEY_NUMBER_ROW, "3"))));
```

Comment:
Integer.parseInt is not necessary if you use the PreferenceManager directly instead of passing through SettingsPreferences

Reply:
ListPreference has psyhical problems wit INT now :/ I was not able to parse INT. I will investigate. https://stackoverflow.com/questions/11346916/listpreference-use-string-array-as-entry-and-integer-array-as-entry-values-does

Reply:
Ok will do it